cmd/main.go

```diff
@@ -33,18 +33,20 @@ import (
 )
 
 var (
-	endpoint                = flag.String("endpoint", "unix:/tmp/csi.sock", "CSI endpoint")
-	nodeID                  = flag.String("nodeid", "", "node id")
-	runController           = flag.Bool("controller", false, "run controller service")
-	runNode                 = flag.Bool("node", false, "run node service")
-	cloudConfigFilePath     = flag.String("cloud-config", "", "Path to GCE cloud provider config")
-	httpEndpoint            = flag.String("http-endpoint", "", "The TCP network address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled.")
-	metricsPath             = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
-	enableMultishare        = flag.Bool("enable-multishare", false, "if set to true, the driver will support multishare instance provisioning")
-	filestoreServiceEnpoint = flag.String("filestore-service-endpoint", "", "Endpoint for filestore service")
-	ecfsDescription         = flag.String("ecfs-description", "", "Filestore multishare instance descrption. ecfs-version=<version>,image-project-id=<projectid>")
-	isRegional              = flag.Bool("is-regional", false, "cluster is regional cluster")
-	gkeClusterName          = flag.String("gke-cluster-name", "", "Cluster Name of the current GKE cluster driver is running on, required for multishare")
+	endpoint                           = flag.String("endpoint", "unix:/tmp/csi.sock", "CSI endpoint")
+	nodeID                             = flag.String("nodeid", "", "node id")
+	runController                      = flag.Bool("controller", false, "run controller service")
+	runNode                            = flag.Bool("node", false, "run node service")
+	cloudConfigFilePath                = flag.String("cloud-config", "", "Path to GCE cloud provider config")
+	httpEndpoint                       = flag.String("http-endpoint", "", "The TCP network address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled.")
+	metricsPath                        = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	enableMultishare                   = flag.Bool("enable-multishare", false, "if set to true, the driver will support multishare instance provisioning")
+	baseFilestoreServiceEndpoint       = flag.String("base-filestore-service-endpoint", "", "Endpoint for filestore service")
+	multiShareFilestoreServiceEndpoint = flag.String("filestore-service-endpoint", "", "Endpoint for multishare filestore service")
```

Comment:
This is non-ideal naming. Ideally the base one could just be called filestore-service-endpoint and the second one was multishare-service-endpoint, but I don't know where to update existing references. If someone could point me to where this flag gets passed in (I'm new to this repo :) ) I am happy to update it and the flag name.

Reply:
you are correct, the ideal naming should have been something like "filestore-service-endpoint" and "filestore-multishare-service-endpoint". 
However, changing that flag name now would be a breaking change, as it's value is plumbed through the GKE control plane.

Filestore [multishare](https://cloud.google.com/filestore/docs/multishares) and single share are variants of Enterprise [tier](https://cloud.google.com/filestore/docs/service-tiers).

I don't forsee a use-case where we need to use different endpoints for different filestore tiers. We can keep it simple and use the same endpoint provided by "filestore-service-endpoint" flags for all filestore tier provisioning.

Reply:
Got it, thanks! Switched to just using the one flag.